modules/task_definition/variables.tf

```diff
@@ -0,0 +1,44 @@
+variable "task_name" {
+  type = string
+}
+
+variable "container_definitions" {}
```

Comment:
Does this want a type?

Reply:
It is a HCL variable, and therefore has no desires.

If you give this a type, it needs to be complete, so you end up typing a complete container_definition (I actually did try that), which then means you can't have optional nulls in that definition easily, which makes a whole mess of complicated code. So this is intentionally untyped.

Reply:
Needs comment to clarify.